rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -68,9 +67,9 @@ namespace rsx
 		case CELL_GCM_CONTEXT_DMA_MEMORY_HOST_BUFFER:
 		case CELL_GCM_LOCATION_MAIN:
 		{
-			if (u32 result = RSXIOMem.RealAddr(offset))
+			if (const u32 ea = render->iomap_table.get_addr(offset); ea + 1)
```

Comment:
Why not just check ea == 0xffffffff? It's more readable that way. Also, the decision to have ea = 0 as legal is dubious, I do not think that is legal for anything but the lv1 kernel itself. I don't mind either way but this add is pointless here for the comparison bit.

Reply:
0 isn't compatoble with the opposite operation (ea->io), but io->ea can still use 0.

Reply:
Also -1 has a cool trick, notice rsx_io_table::get_addr no longer checks for unmapped memory.
Thats because OR doesnt change -1, unlike 0.

Reply:
That's fine, but the comparison for invalid still irks me. It relies on arithmetic to infer a hard-coded constant which is messy. Can this be done in the table class with proper named constants? I think a static constexpr may work for this. The reason 0 comparison gets a pass is because it is a universal 'false' since forever, hence the special handling even in the language itself.
Alternatively get_addr may return a pair, but that is overkill.